pkg/gauges/contracts/BalancerMinter.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeMath.sol";
+
+import "./interfaces/IBalancerToken.sol";
+import "./interfaces/IGaugeController.sol";
+import "./interfaces/ILiquidityGauge.sol";
+
+contract BalancerMinter {
+    using SafeMath for uint256;
+
+    IBalancerToken private immutable _token;
+    IGaugeController private immutable _controller;
```

Comment:
I've added some getters for these values however I've renamed these variables in order to not try to be ABI compatible.
This is as if we exposed `_token` with a `token()` getter it may create confusion as it's pointing towards the `BalancerTokenAdmin` contract which is not a token.
 
 I doubt that any contracts rely on polling the minter contract for the CRV or Gauge Controller address so it's best to optimise for clarity over compatibility.

Reply:
I see, that's a fair point. If anyone did call `token` then they'd be making a mistake in our case. I like your reasoning :+1: